### PR TITLE
MVP

### PR DESCRIPTION
**Pull request by jspaezp**
Makes this an MVP

Works pretty decently If I say so myself.

### CODE REVIEW

File: tdf_simulator/config.py [hunk 1]
```diff
@@ -148,8 +189,12 @@ def from_dict(cls, d: dict[str, Any]) -> WindowInfo:
             collision_energy=d["CollisionEnergy"],
         )
 
+    def to_dict(self) -> dict:
+        return asdict(self)
```
Comment (sander-willems-bruker) [inline tdf_simulator/config.py]: This does not work properly as names as incorrect. Quick fix is to replace with:
```python
        return {
            "WindowGroup": self.window_group,
            "ScanNumBegin": self.scan_num_begin,
            "ScanNumEnd": self.scan_num_end,
            "IsolationMz": self.isolation_mz,
            "IsolationWidth": self.isolation_width,
            "CollisionEnergy": self.collision_energy,
        }
```

**Comment by sander-willems-bruker** [comment]
Note that your testing relies on timsrust_pyo3, which still uses TimsRust 0.2.3. This is severely outdated by now and some errors pop-up (because prevously diaFrames were not fully implemented yet)

**Comment by sander-willems-bruker** [comment]
Th number of peaks as reported in "Frames" is calculated incorrectly. Will investigate a little.

EDIT:
Update lines 164 and 415 of `transition_simulator.py` from `npeaks = len(out.frame_ints)` to `npeaks = out.num_peaks`

**Comment by sander-willems-bruker** [comment]
Havent checked out the implementation details, but what I tried out as an end-user seems to work as expected, awesome job!

**Comment by jspaezp** [comment]
> Note that your testing relies on timsrust_pyo3, which still uses TimsRust 0.2.3. This is severely outdated by now and some errors pop-up (because prevously diaFrames were not fully implemented yet)

yeah the 0.4.0 migration is in my checklist for next week!

**Comment by jspaezp** [comment]
> Th number of peaks as reported in "Frames" is calculated incorrectly. Will investigate a little.
> EDIT:
> Update lines 164 and 415 of transition_simulator.py from npeaks = len(out.frame_ints) to npeaks = out.num_peaks

nice catch! thanks a lot!

**Pull request merged**